### DISCUSSION TITLE
hardware sensors

### DISCUSSION THREAD

**Discussion by wiltet** [mathoudebine/turing-smart-screen-python#205] [Ideas]
Hi,
first I want to thank you for this great project and all the work you spent on it.

These turing displays come more and more famous because they are affordable, economic and good quality displays. As well I do like the idea of open software projects, specially when there is creativity involved in to the use of the product.

Im a bit disillusioned about the vendors software, that they dont implement the possibility to grab sensor datas from HWinfo, AIDA or others.
Is there a opportunity for you to implement this into your project? I think Martin from HWinfo would certainly like to help you realising this.

thanks and stay tuned

**Comment by mathoudebine** [comment]
Hi and thanks for your message. I'm really glad to see these displays being used for completely different purpose like smart home, notifications, public transport schedules...

I think the main issue with HWiNFO is that the Shared Memory support for HWiNFO64 is disabled every 12 hours and require manual action to enable it again. This Shared Memory support is the interface between HWiNFO and 3rd party software like this one, so for me it is a major blocking point.
Also, the user must have HWiNFO running in background in addition to this program.
Finally the SDK is not publicly available, so it would prevent the community to maintain and evolve the HWiNFO support for this project, particularly for forks.
There are similar issued with Aida64 (and the software is not free).

This is why we choose to go with [LibreHardwareMonitor](https://github.com/LibreHardwareMonitor/LibreHardwareMonitor) library for HW sensor support on Windows platforms: the library has similar capabilities to HWiNFO but it is open-source and can be distributed with this software. It does not require the user to run another program in background, and it can be tested in automatic builds on Github CI.

If someone implements the support for HWiNFO / Aida64 i would be happy to add it to the project! But i will not be able to maintain it myself

**Comment by wiltet** [comment]
Thanks for your honest statement.

Unfortunately I cannot help with the programming, otherwise I would have already started.
